### ISSUE TITLE
Space character is being lost in Subject header value

### ISSUE THREAD

**Issue by titarenko** [nodemailer/buildmail#11]
Hi!

Using method `encodeWords` [here](https://github.com/nodemailer/buildmail/blob/master/lib/buildmail.js#L997) causes space character (at random place, depending on content length, I guess) to be lost.

So for mail with subject:

```js
var mail = { subject: 'Бета-продакшн: платежка 2422750 от 2015-07-20' };
```

With header subject value resulting in (before sending over the wire):

=?UTF-8?Q?=D0=91=D0=B5=D1=82=D0=B0-=D0=BF=D1=80?= =?UTF-8?Q?=D0=BE=D0=B4=D0=B0=D0=BA=D1=88=D0=BD=3A_?= =?UTF-8?Q?=D0=BF=D0=BB=D0=B0=D1=82=D0=B5=D0=B6=D0=BA?= =?UTF-8?Q?**=D0=B0?= 2**422750 =?UTF-8?Q?=D0=BE=D1=82?= 2015-07-20

Client (tested with gmail.com, mail.yandex.ua) will got message with subject header value (after receiving over the wire):

=?utf-8?Q?=D0=91=D0=B5=D1=82=D0=B0-=D0=BF=D1=80=D0=BE=D0=B4=D0=B0=D0=BA=D1=88=D0=BD:=20=D0=BF=D0=BB=D0=B0=D1=82=D0=B5=D0=B6=D0=BA**=D0=B02**422750=20=D0=BE=D1=82=202015-07-20?=

Resulting in subject displayed as:

Бета-продакшн: платежк**а2**422750 от 2015-07-20

That's pretty frustrating. But could be avoided by encoding subject as complete thing using `encodeWord` method with same arguments as `encodeWords` (tested by ad-hoc modification of node_modules code base in place on my server).

I guess, that happens, because some server in the chain between sender and receiver sees space between 2 encoded entities then ignores it and concatenates encoded entities. But if whole string will be encoded as 1 entity, that does not happen. And `encodeWord` is probably the best choice to encode whole header value as single entity, and not as a set of space-delimited words, which can be wrongly processed by email servers.

Could you check this issue (originally related to using `nodemailer` library)?

Thanks!

**Comment by andris9** [comment]
It's an interesting problem. Are you using some of the well known SMTP providers like SendGrid or Mailgun? Or are you running your own SMTP MTA? This issue most probably happens when a message is rewritten by a MTA (for example if the MTA adds tracking links etc.)

**Comment by titarenko** [comment]
We are using mailchimp.

**Comment by andris9** [comment]
Thanks, I'll try to replicate this myself to see what and where gets broken

**Comment by andris9** [comment]
I verified the bug and this is an issue in Mandrill and not in Nodemailer. I contacted Mandrill support and explained what is going on. The actual problem is in how Mandrill treats header line folding.

Nodemailer sends out Subject header line folded like this:

```
Subject: =?UTF-8?Q?=D0=B0?=<CR><LF>
<SP>2
```

Mandrill de-folds by removing all whitespace symbols and ends up with something like this:

```
Subject: =?UTF-8?Q?=D0=B0?=2
```

While it actually should have this:

```
Subject: =?UTF-8?Q?=D0=B0?=<SP>2
```

Anyway, I'm waiting for some kind of response from Mandrill and decide how to handle this based on whatever they say.
